### ISSUE TITLE
gpt-oss sft开启router_aux_loss_coef loss异常

### ISSUE THREAD

**Issue by liuqianchao** [modelscope/ms-swift#5627]
**Describe the bug**
gpt-oss sft训练开启router_aux_loss_coef之后loss异常：
router_aux_loss_coef = 0.9

{'loss': 930.53393555, 'grad_norm': 143.61636353, 'learning_rate': 5.3e-07, 'aux_loss': 4.01051664, 'token_acc': 0.22081266, 'epoch': 0.01, 'global_step/max_steps': '1/369', 'percentage': '0.27%', 'elapsed_time': '7m 11s', 'remaining_time': '1d 20h 4m 17s', 'memory(GiB)': 26.38, 'train_speed(iter/s)': 0.002319}
{'loss': 930.2043457, 'grad_norm': 107.51234436, 'learning_rate': 2.63e-06, 'aux_loss': 4.01045179, 'token_acc': 0.23505743, 'epoch': 0.04, 'global_step/max_steps': '5/369', 'percentage': '1.36%', 'elapsed_time': '33m 57s', 'remaining_time': '1d 17h 12m 14s', 'memory(GiB)': 26.39, 'train_speed(iter/s)': 0.002454}
{'loss': 927.21035156, 'grad_norm': 20.89083481, 'learning_rate': 5.26e-06, 'aux_loss': 4.00914497, 'token_acc': 0.46115847, 'epoch': 0.08, 'global_step/max_steps': '10/369', 'percentage': '2.71%', 'elapsed_time': '1h 8m 56s', 'remaining_time': '1d 17h 15m 2s', 'memory(GiB)': 26.39, 'train_speed(iter/s)': 0.002417}
{'loss': 924.36416016, 'grad_norm': 4.26955414, 'learning_rate': 7.89e-06, 'aux_loss': 4.00620689, 'token_acc': 0.73503175, 'epoch': 0.12, 'global_step/max_steps': '15/369', 'percentage': '4.07%', 'elapsed_time': '1h 45m 46s', 'remaining_time': '1d 17h 36m 9s', 'memory(GiB)': 26.39, 'train_speed(iter/s)': 0.002364}
{'loss': 923.41845703, 'grad_norm': 2.00485945, 'learning_rate': 1e-05, 'aux_loss': 4.00486412, 'token_acc': 0.82651183, 'epoch': 0.16, 'global_step/max_steps': '20/369', 'percentage': '5.42%', 'elapsed_time': '2h 23m 1s', 'remaining_time': '1d 17h 35m 40s', 'memory(GiB)': 26.4, 'train_speed(iter/s)': 0.002331}

如果router_aux_loss_coef = 0
{'loss': 6.51318359, 'grad_norm': 144.81030273, 'learning_rate': 2e-06, 'token_acc': 0.21845639, 'epoch': 0.03, 'global_step/max_steps': '1/93', 'percentage': '1.08%', 'elapsed_time': '7m 0s', 'remaining_time': '10h 44m 58s', 'memory(GiB)': 26.38, 'train_speed(iter/s)': 0.002377}
{'loss': 5.0801239, 'grad_norm': 37.58584595, 'learning_rate': 1e-05, 'token_acc': 0.32137252, 'epoch': 0.16, 'global_step/max_steps': '5/93', 'percentage': '5.38%', 'elapsed_time': '34m 50s', 'remaining_time': '10h 13m 8s', 'memory(GiB)': 30.35, 'train_speed(iter/s)': 0.002392}
{'loss': 1.34742126, 'grad_norm': 2.82086992, 'learning_rate': 9.92e-06, 'token_acc': 0.73217829, 'epoch': 0.33, 'global_step/max_steps': '10/93', 'percentage': '10.75%', 'elapsed_time': '1h 12m 31s', 'remaining_time': '10h 1m 54s', 'memory(GiB)': 30.35, 'train_speed(iter/s)': 0.002298}
{'loss': 0.60466156, 'grad_norm': 1.12729561, 'learning_rate': 9.68e-06, 'token_acc': 0.84275398, 'epoch': 0.49, 'global_step/max_steps': '15/93', 'percentage': '16.13%', 'elapsed_time': '1h 50m 45s', 'remaining_time': '9h 35m 59s', 'memory(GiB)': 30.35, 'train_speed(iter/s)': 0.002257}
{'loss': 0.47910767, 'grad_norm': 0.79875964, 'learning_rate': 9.3e-06, 'token_acc': 0.86584298, 'epoch': 0.65, 'global_step/max_steps': '20/93', 'percentage': '21.51%', 'elapsed_time': '2h 29m 9s', 'remaining_time': '9h 4m 24s', 'memory(GiB)': 30.35, 'train_speed(iter/s)': 0.002235}

这里哪里可能会有问题？

**Comment by Jintao-Huang** [comment]
直接用的transformers实现

**Comment by liuqianchao** [comment]
@Jintao-Huang 可能是代码里loss计算的batch size没对齐导致的？

https://github.com/modelscope/ms-swift/blob/e15ed0196a14414d4a29e27e923d856385869b0e/swift/trainers/trainers.py#L299

原始loss已经处过了batch_size，但加的moe loss没有处理？

<img width="2154" height="710" alt="Image" src="https://github.com/user-attachments/assets/8e9ec011-8b7a-4c7e-a665-6770f21b93d5" />

**Comment by liuqianchao** [comment]
试了一下把训练实际的batch size调整成1，就没有这个问题了

**Comment by Jintao-Huang** [comment]
训练脚本是什么发一下

**Comment by Jintao-Huang** [comment]
已经复现了
